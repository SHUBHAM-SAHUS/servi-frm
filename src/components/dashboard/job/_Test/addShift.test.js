import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import AddShift from '../addShift';

describe('AddShift Component Test', () => {
    const initialState = {
        sAJobCalendar: {
            siteSupervisorsList: [],
            supervisorsList: []
        },
        jobdocManagement: {
            orgUSerList: []
        },
        form: {
            addShift: {}
        }
    }

    const mockStore = configureStore()
    let store, container

    beforeEach(() => {
        store = mockStore(initialState)
        container = shallow(<AddShift store={store} />)
    })

    it('should render component AddShift correctly', () => {
        const component = shallow(<AddShift store={store} />);
        expect(component).toMatchSnapshot();
    });

    it('Render the connected(SMART) AddShift component', () => {
        expect(container.length).toEqual(1)
    });
})