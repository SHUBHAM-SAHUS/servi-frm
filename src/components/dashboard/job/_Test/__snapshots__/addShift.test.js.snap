// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddShift Component Test should render component AddShift correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UNDEFINED
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="addShift"
        formValues={Object {}}
        getFormState={[Function]}
        jobCalendarActions={
          Object {
            "addTaskDetails": [Function],
            "getAllServiceAgentJobs": [Function],
            "getServiceAgentStaffMembers": [Function],
            "getSiteSupervisorsList": [Function],
            "getSupervisorsList": [Function],
            "getTaskDetails": [Function],
            "updateTaskDetails": [Function],
          }
        }
        keepDirtyOnReinitialize={false}
        orgUSerList={Array []}
        persistentSubmitErrors={false}
        pure={true}
        sAJobMgmtAction={
          Object {
            "addJobSWMSSign": [Function],
            "deleteJobScheduleShift": [Function],
            "deletePhotosDocsFile": [Function],
            "deleteReport": [Function],
            "getJobDetails": [Function],
            "getJobMembers": [Function],
            "getJobReport": [Function],
            "getJobSignOff": [Function],
            "getPhotosDocsFileList": [Function],
            "getSWMSSignDetails": [Function],
            "getSingleUserLicences": [Function],
            "lockSignatureProcess": [Function],
            "saveJobReport": [Function],
            "sendEmailJobReport": [Function],
            "signOffJobSheet": [Function],
            "updateJob": [Function],
            "updateJobReport": [Function],
            "updateJobScheduleShift": [Function],
            "updateUserSiteTime": [Function],
          }
        }
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        siteSupervisorsList={Array []}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        submitAsSideEffect={false}
        supervisorsList={Array []}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />,
      "value": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "addShift",
        "formValues": Object {},
        "getFormState": [Function],
        "jobCalendarActions": Object {
          "addTaskDetails": [Function],
          "getAllServiceAgentJobs": [Function],
          "getServiceAgentStaffMembers": [Function],
          "getSiteSupervisorsList": [Function],
          "getSupervisorsList": [Function],
          "getTaskDetails": [Function],
          "updateTaskDetails": [Function],
        },
        "keepDirtyOnReinitialize": false,
        "orgUSerList": Array [],
        "persistentSubmitErrors": false,
        "pure": true,
        "sAJobMgmtAction": Object {
          "addJobSWMSSign": [Function],
          "deleteJobScheduleShift": [Function],
          "deletePhotosDocsFile": [Function],
          "deleteReport": [Function],
          "getJobDetails": [Function],
          "getJobMembers": [Function],
          "getJobReport": [Function],
          "getJobSignOff": [Function],
          "getPhotosDocsFileList": [Function],
          "getSWMSSignDetails": [Function],
          "getSingleUserLicences": [Function],
          "lockSignatureProcess": [Function],
          "saveJobReport": [Function],
          "sendEmailJobReport": [Function],
          "signOffJobSheet": [Function],
          "updateJob": [Function],
          "updateJobReport": [Function],
          "updateJobScheduleShift": [Function],
          "updateUserSiteTime": [Function],
        },
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "siteSupervisorsList": Array [],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "submitAsSideEffect": false,
        "supervisorsList": Array [],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "destroyOnUnmount": true,
          "enableReinitialize": false,
          "forceUnregisterOnUnmount": false,
          "form": "addShift",
          "getFormState": [Function],
          "keepDirtyOnReinitialize": false,
          "persistentSubmitErrors": false,
          "pure": true,
          "shouldAsyncValidate": [Function],
          "shouldError": [Function],
          "shouldValidate": [Function],
          "shouldWarn": [Function],
          "submitAsSideEffect": false,
          "touchOnBlur": true,
          "touchOnChange": false,
          "updateUnregisteredFields": false,
        },
        "displayName": "ReduxForm",
        "render": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          destroyOnUnmount={true}
          enableReinitialize={false}
          forceUnregisterOnUnmount={false}
          form="addShift"
          formValues={Object {}}
          getFormState={[Function]}
          jobCalendarActions={
            Object {
              "addTaskDetails": [Function],
              "getAllServiceAgentJobs": [Function],
              "getServiceAgentStaffMembers": [Function],
              "getSiteSupervisorsList": [Function],
              "getSupervisorsList": [Function],
              "getTaskDetails": [Function],
              "updateTaskDetails": [Function],
            }
          }
          keepDirtyOnReinitialize={false}
          orgUSerList={Array []}
          persistentSubmitErrors={false}
          pure={true}
          sAJobMgmtAction={
            Object {
              "addJobSWMSSign": [Function],
              "deleteJobScheduleShift": [Function],
              "deletePhotosDocsFile": [Function],
              "deleteReport": [Function],
              "getJobDetails": [Function],
              "getJobMembers": [Function],
              "getJobReport": [Function],
              "getJobSignOff": [Function],
              "getPhotosDocsFileList": [Function],
              "getSWMSSignDetails": [Function],
              "getSingleUserLicences": [Function],
              "lockSignatureProcess": [Function],
              "saveJobReport": [Function],
              "sendEmailJobReport": [Function],
              "signOffJobSheet": [Function],
              "updateJob": [Function],
              "updateJobReport": [Function],
              "updateJobScheduleShift": [Function],
              "updateUserSiteTime": [Function],
            }
          }
          shouldAsyncValidate={[Function]}
          shouldError={[Function]}
          shouldValidate={[Function]}
          shouldWarn={[Function]}
          siteSupervisorsList={Array []}
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
          submitAsSideEffect={false}
          supervisorsList={Array []}
          touchOnBlur={true}
          touchOnChange={false}
          updateUnregisteredFields={false}
        />,
        "value": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "destroyOnUnmount": true,
          "enableReinitialize": false,
          "forceUnregisterOnUnmount": false,
          "form": "addShift",
          "formValues": Object {},
          "getFormState": [Function],
          "jobCalendarActions": Object {
            "addTaskDetails": [Function],
            "getAllServiceAgentJobs": [Function],
            "getServiceAgentStaffMembers": [Function],
            "getSiteSupervisorsList": [Function],
            "getSupervisorsList": [Function],
            "getTaskDetails": [Function],
            "updateTaskDetails": [Function],
          },
          "keepDirtyOnReinitialize": false,
          "orgUSerList": Array [],
          "persistentSubmitErrors": false,
          "pure": true,
          "sAJobMgmtAction": Object {
            "addJobSWMSSign": [Function],
            "deleteJobScheduleShift": [Function],
            "deletePhotosDocsFile": [Function],
            "deleteReport": [Function],
            "getJobDetails": [Function],
            "getJobMembers": [Function],
            "getJobReport": [Function],
            "getJobSignOff": [Function],
            "getPhotosDocsFileList": [Function],
            "getSWMSSignDetails": [Function],
            "getSingleUserLicences": [Function],
            "lockSignatureProcess": [Function],
            "saveJobReport": [Function],
            "sendEmailJobReport": [Function],
            "signOffJobSheet": [Function],
            "updateJob": [Function],
            "updateJobReport": [Function],
            "updateJobScheduleShift": [Function],
            "updateUserSiteTime": [Function],
          },
          "shouldAsyncValidate": [Function],
          "shouldError": [Function],
          "shouldValidate": [Function],
          "shouldWarn": [Function],
          "siteSupervisorsList": Array [],
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "submitAsSideEffect": false,
          "supervisorsList": Array [],
          "touchOnBlur": true,
          "touchOnChange": false,
          "updateUnregisteredFields": false,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "destroyOnUnmount": true,
            "enableReinitialize": false,
            "forceUnregisterOnUnmount": false,
            "form": "addShift",
            "getFormState": [Function],
            "keepDirtyOnReinitialize": false,
            "persistentSubmitErrors": false,
            "pure": true,
            "shouldAsyncValidate": [Function],
            "shouldError": [Function],
            "shouldValidate": [Function],
            "shouldWarn": [Function],
            "submitAsSideEffect": false,
            "touchOnBlur": true,
            "touchOnChange": false,
            "updateUnregisteredFields": false,
          },
          "displayName": "ReduxForm",
          "render": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
